Added test for smart graph vertex validator.

adityamukho opened the pull request:
Adds a test for SmartVertex schema validator, which will be introduced in 3.10.0.

**Note:**
This PR can only be merged AFTER the following PRs are merged into `devel`. Otherwise we expect 3.9.* -> 3.10.* Jenkins run to always fail.

https://github.com/arangodb/arangodb/pull/15090
https://github.com/arangodb/enterprise/pull/798

dothebart reviewed (commented):
please note that
- you must revalidate the supported versions
- if you write in checkdata you must clean up, since it can be run multiple times
- you may be invoked on a read only cluster

adityamukho commented:
1. Added specific check for current version >= 3.10.0.
2. Added check for readOnly cluster. Test is ignored in that case.
3. If anything is written in checkdata, it automatically fails the test. Hence, no cleanup required.

hkernbach commented:
Related BTS Ticket: https://arangodb.atlassian.net/browse/BTS-663
